ui/src/App.tsx

```diff
@@ -45,46 +45,76 @@ function App() {
     id: '',
   });
   const [namespace, setNamespace] = useState('');
+  const [namespaces, setNamespaces] = useState<string[]>([]);
   const [tokensDisabled, setTokensDisabled] = useState(false);
   const [blockchainPlugin, setBlockchainPlugin] = useState('');
   const [tutorialSections, setTutorialSections] = useState<ITutorialSection[]>(
     []
   );
 
+  const changeNamespace = (ns: string) => {
+    if (ns !== namespace) {
+      // trigger refresh
+      localStorage.setItem('sandboxNamespace', ns);
+      window.location.reload();
+    }
+  };
+
   useEffect(() => {
-    Promise.all([
-      fetchCatcher(`${SDK_PATHS.status}`),
-      fetchCatcher(`${SDK_PATHS.organizations}/self`),
-      fetchCatcher(`${SDK_PATHS.plugins}`),
-    ])
-      .then(async ([statusResponse, orgResponse, pluginsResponse]) => {
-        (!pluginsResponse.tokens || pluginsResponse.tokens.length === 0) &&
-          setTokensDisabled(true);
-        setBlockchainPlugin(
-          pluginsResponse.blockchain.length > 0
-            ? pluginsResponse.blockchain[0].pluginType
-            : ''
+    fetchCatcher(`${SDK_PATHS.namespaces}`)
+      .then(async (namespacesResponse) => {
+        const ffNamespaces = namespacesResponse as IFireflyNamespace[];
+        let selectedNamespace: IFireflyNamespace | undefined;
+        // load previous selected namespace from browser cache
+        const savedSelectedNamespaceName =
```

Comment:
I think we should transition from using local storage directly, to using jotai's `atomWithStorage` https://jotai.org/docs/utils/atom-with-storage